src/rez/utils/pip.py

```diff
@@ -0,0 +1,689 @@
+    """Convert a distribution name to a rez compatible name.
+
+    The rez package name can't be simply set to the dist name, because some
+    pip packages have hyphen in the name. In rez this is not a valid package
+    name (it would be interpreted as the start of the version).
+
+    Example: my-pkg-1.2 is 'my', version 'pkg-1.2'.
+
+    Args:
+        dist_name (str): Distribution name to convert.
+
+    Returns:
+        str: Rez-compatible package name.
+    """
+    return dist_name.replace("-", "_")
+
+
+def pip_to_rez_version(dist_version, allow_legacy=True):
+    """Convert a distribution version to a rez compatible version.
+
+    TODO [AJ] needs a table of example conversions.
+
+    The python version schema specification isn't 100% compatible with rez.
+
+    1: version epochs (they make no sense to rez, so they'd just get stripped
+       of the leading N!;
+    2: python versions are case insensitive, so they should probably be
+       lowercased when converted to a rez version.
+    3: local versions are also not compatible with rez
+
+    The canonical public version identifiers MUST comply with the following scheme:
+    [N!]N(.N)*[{a|b|rc}N][.postN][.devN]
+
+    Epoch segment: N! - skip
+    Release segment: N(.N)* 0 as is
+    Pre-release segment: {a|b|c|rc|alpha|beta|pre|preview}N - always lowercase
+    Post-release segment: .{post|rev|r}N - always lowercase
+    Development release segment: .devN - always lowercase
+
+    Local version identifiers MUST comply with the following scheme:
+    <public version identifier>[+<local version label>] - use - instead of +
+
+    Args:
+        dist_version (str): The distribution version to be converted.
+        allow_legacy (bool): Flag to allow/disallow PEP440 incompatibility.
+
+    Returns:
+        str: Rez-compatible equivalent version string.
+
+    Raises:
+        InvalidVersion: When legacy mode is not allowed and a PEP440
+        incompatible version is detected.
+
+    .. _PEP 440 (all possible matches):
+        https://www.python.org/dev/peps/pep-0440/#appendix-b-parsing-version-strings-with-regular-expressions
+
+    .. _Core utilities for Python packages:
+        https://packaging.pypa.io/en/latest/version/
+
+    """
+    pkg_version = packaging_parse(dist_version)
+
+    if isinstance(pkg_version, packaging_LegacyVersion):
+        if allow_legacy:
+            print_warning(
+                "Invalid PEP440 version detected: %r. Reverting to legacy mode.",
+                pkg_version
+            )
+            # this will always be the entire version string
+            return pkg_version.base_version.lower()
+        else:
+            raise packaging_InvalidVersion(
+                "Version: {} is not compatible with PEP440.".format(dist_version)
+            )
+
+    # the components of the release segment excluding epoch or any
+    # prerelease/development/postrelease suffixes
+    rez_version = pkg_version.base_version
```

Comment:
Umm, I'm sorry but that introduces a bug. I thought the base version would be enough, but it includes the epoch... So we will need to go back to the previous version of the code. Sorry about that.